### ISSUE TITLE
PHP 8.1 - Too few arguments for mysqli_get_client_info

### ISSUE THREAD

**Issue by williamdes** [vimeo/psalm#5852]
```
ERROR: TooFewArguments - libraries/classes/Dbal/DbiMysqli.php:407:16 - Too few arguments for mysqli_get_client_info - expecting 1 but saw 0 (see https://psalm.dev/025)
        return mysqli_get_client_info();
```

See: https://github.com/phpmyadmin/phpmyadmin/issues/16911
See: https://github.com/php/php-src/commit/7e9f6d2a48f8ff6ae458252d395eec1b1d9e6f14

On PHP 8.1+ mysqli_get_client_info does not need arguments.

**Comment by williamdes** [comment]
https://psalm.dev/r/0c484fe235

See: https://github.com/phpmyadmin/phpmyadmin/runs/2694551783#step:7:18

Note: please change the output format for GitHub, the links to psalm end up in 404 because it picks `)` as part of the link

Maybe update: https://psalm.dev/025

**Comment by orklah** [comment]
PHP 8.1 is not yet handled in callmaps: https://github.com/vimeo/psalm/tree/master/dictionaries

Feel free to PR if you want to speed up the support of 8.1 in Psalm

**Comment by williamdes** [comment]
More refs can be found on https://github.com/phpstan/phpstan-src/pull/545

**Comment by orklah** [comment]
Oh then if it's from PHP 8.0, everything is already in place for making the change

**Comment by williamdes** [comment]
> Oh then if it's from PHP 8.0, everything is already in place for making the change

Only 8.1+
Ref: https://php.watch/versions/8.1/mysqli_get_client_info-deprecation

**Comment by orklah** [comment]
well, there's multiple layers here and I'm not sure which php version is concerned
1) mysqli_get_client_info function. It is documented in Psalm with a mandatory param of type mysqli
- From which php version did the function accept not taking a parameter?
- PHP 8.1 will deprecate pushing a parameter but it will still work
- PHP 8.2 or maybe event PHP 9 will remove the possibility to push a parameter, at which point, the parameter will have to be removed from Psalm

2) mysqli::get_client_info method. It is documented in psalm without parameters
- PHP 8.1 will the method but it will still work
- PHP 8.2 or maybe event PHP 9 will remove the method, at which point, it will have to be removed from Psalm


So, for now, the method doesn't have to change. Likewise, PHP 8.1 will deprecate the parameter of the function but apparently, it already worked without a parameter before PHP 8.1 (and this is what we need to know in order to fix the stubs in psalm)

**Comment by villfa** [comment]
The signature of *mysqli_get_client_info()* has been fixed by #7186
